orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -397,16 +671,22 @@ Status TrainingRunner::TrainingLoop(IDataLoader& training_data_loader, IDataLoad
     epoch++;
   }
 
-  std::cout << "Round: " << round_ << "\n"
-            << "Batch size: " << params_.batch_size << "\n"
-            << "Number of Batches: " << (step_ - step_start) << "\n"
-            << "Gradient Accumulation Steps: " << gradient_accumulation_step_count << "\n"
-            << "Weight Update Steps: " << (weight_update_step_count_ - weight_update_step_count_start) << "\n"
-            << "Total Running Time: " << total_time << " Seconds \n"
-            << "Average Running Time Per Batch: " << total_time / (step_ - step_start) * 1000 << " ms\n"
-            << "Throughput: " << params_.batch_size * (step_ - step_start) / total_time << " Examples / Second\n"
-            << "Stabilized Throughput: " << params_.batch_size / (stabilized_total_time / stabilized_perf_total_step_count)
-            << " Examples / Second\n";
+  // Print some info when reaching the end of the batch.
+  // For pipeline, the first stage, idexed by 0, computes the start and end of each batch.
+  // That is, it computes the start and the end even for the last, and we threfore should
+  // print out timing result at the first stage.
+  if (pipeline_context_.pipeline_stage_id == 0) {
```

Comment:
>pipeline_context_.pipeline_stage_id == 0 [](start = 6, length = 40)

better to let MPI to filter messages from different ranks

Reply:
Ok.